web/client-api/src/main/java/io/deephaven/web/client/api/QueryConnectable.java

```diff
@@ -246,12 +244,8 @@ public void disconnected() {
 
     public abstract void notifyServerShutdown(TerminationNotificationResponse success);
 
-    public boolean useWebsockets() {
-        Boolean useWebsockets = getOptions().useWebsockets;
-        if (useWebsockets == null) {
-            useWebsockets = getServerUrl().startsWith("http:");
-        }
-        return useWebsockets;
+    public boolean supportsClientStreaming() {
+        return getOptions().transportFactory.getSupportsClientStreaming();
```

Comment:
While it's correct since the `transportFactory` is always initialized correctly by `IdeConnection`, in `ConnectionOptions` it annotates `transportFactory` as `JsNullable`. I guess we'd just throw here if it was null anyways?

Reply:
Yes - the annotations are for the sake of the generated TS and docs ("you can populate this or not"). In theory the compiler could take them seriously, but in practice Java isn't expressive enough about nulls, even with this annotation (static analysis is used instead, to solve for fields/variables that can never be null).